packages/eslint-plugin/src/rules/member-ordering.ts

```diff
@@ -397,6 +397,16 @@ function getRankOrder(
   return rank;
 }
 
+function getAccessibility(node: Member): TSESTree.Accessibility {
+  if ('accessibility' in node && node.accessibility) {
+    return node.accessibility;
+  }
+  if ('key' in node && node.key?.type === AST_NODE_TYPES.PrivateIdentifier) {
+    return 'private';
```

Comment:
I don't think that `private` an `#private` should be treated the same.
Syntactically AND semantically they're very different so grouping them together is incorrect IMO.

Reply:
Do you want to add a new group to sort such fields? What would you call it?

Reply:
TSESTree.Accessibility is 'private' | 'protected' | 'public'; in terms of accessibility in Typescript, `private` and `#private` are no different.

Reply:
> TSESTree.Accessibility is 'private' | 'protected' | 'public'; in terms of accessibility in Typescript, private and #private are no different.

That is incorrect.

The string union which we use in the AST had those three values, but the [`accessibility` property on the AST](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/ast-spec/src/base/PropertyDefinitionBase.ts#L21) has four states as it is optional.

For a `private` property, `node.accessibility === 'private'`, but for a `#private` property, `node.accessibility === undefined`. So the AST represents them differently there.
Additionally a `private` property can have either an identifier (`private identifier`), a string (`private 'string') or a computed key (`private [someExpression]`). However a `#private` property can only have an identifier proceeded by a `#` (called a `PrivateIdentifier`).

So there are many differences in how the AST is represented between the two.

And in terms of semantics - `private` properties are just semantically hidden in TS checks, but are fully accessible to any code via casts or `@ts-ignore`s. They're also validly accessible via index access (`instance['privateProp']`) in any code without a TS error.
OTOH a `#private` property is invisible and inaccessible outside the class. It is literally impossible for anyone to access a `#private` property outside the class.

Reply:
> Do you want to add a new group to sort such fields? What would you call it?

Yes please! A new group would let people sort `#private` properties into their own group of they wish or merge them with another group.

As for naming, I'm not sure. Maybe just copy the `private` naming but add a `#` to it? So like `#private-field`, `#private-method`, etc.

I think that might be the clearest naming we can use. What do you think?

Reply:
@sviat9440 we're waiting on a response to this 🙂

Reply:
fixed